nyan/value/value.cpp

```diff
@@ -124,6 +124,18 @@ static void handle_modifiers(const std::vector<Type> &modifiers,
                              const ValueHolder &value_holder,
 							 std::vector<std::pair<fqon_t, Location>> *objs_in_values) {
 	bool contains_optional = false;
+	for (auto &mod: modifiers) {
+		auto modifier_type = mod.get_composite_type();
+
+		if (modifier_type == composite_t::OPTIONAL) {
+			contains_optional = true;
+		}
```

Comment:
interesting corner case, which `optional` do we assign to in `optional(optional(int)) = None` (it won't matter in practice since the returnvalue is None in either way)

Reply:
It is just assigned to the member because `optional` does not have its own `Value` instance. In the current implementation `optional(optional(int))` is treated the same as `optional(int)`.

Reply:
ok that resolves that, but what about `optional(bool) = None`? we have do decide on a behavior there, i'd say we try to assign it from the inside to the outside until the operator `=` is accepted. then we also don't need all the none-assignments in all the types